ray-operator/test/e2e/rayservice_in_place_update_test.go

```diff
@@ -0,0 +1,113 @@
+package e2e
+
+import (
+	"path"
+	"strings"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	"github.com/ray-project/kuberay/ray-operator/test/sampleyaml"
+	. "github.com/ray-project/kuberay/ray-operator/test/support"
+)
+
+func TestRayServiceInPlaceUpdate(t *testing.T) {
+	test := With(t)
+	g := NewWithT(t)
+
+	// Create a namespace
+	namespace := test.NewTestNamespace()
+	test.StreamKubeRayOperatorLogs()
+
+	fileName := "ray-service.sample.yaml"
+
+	yamlFilePath := path.Join(sampleyaml.GetSampleYAMLDir(test), fileName)
+	rayServiceFromYaml := DeserializeRayServiceYAML(test, yamlFilePath)
+	KubectlApplyYAML(test, yamlFilePath, namespace.Name)
+
+	rayService, err := GetRayService(test, namespace.Name, rayServiceFromYaml.Name)
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(rayService).NotTo(BeNil())
+
+	var rayClusterName string
+	// Wait for RayCluster name to be populated
+	g.Eventually(UnderlyingRayCluster(test, rayService), TestTimeoutShort).Should(
+		WithTransform(func(cluster *rayv1.RayCluster) string {
+			rayClusterName = cluster.Name
+			return rayClusterName
+		}, Not(BeEmpty())),
+	)
+
+	test.T().Logf("Waiting for RayCluster %s/%s to be ready", namespace.Name, rayClusterName)
+	g.Eventually(RayCluster(test, namespace.Name, rayClusterName), TestTimeoutMedium).
+		Should(WithTransform(RayClusterState, Equal(rayv1.Ready)))
+	rayCluster, err := GetRayCluster(test, namespace.Name, rayClusterName)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	// Create curl pod
+	curlPodName := "curl-pod"
+	curlContainerName := "curl-container"
+
+	curlPod, err := CreateCurlPod(test, curlPodName, curlContainerName, namespace.Name)
+	g.Expect(err).NotTo(HaveOccurred())
+	// Wait until curl pod is created
+	g.Eventually(func(g Gomega) *corev1.Pod {
+		updatedCurlPod, err := test.Client().Core().CoreV1().Pods(curlPod.Namespace).Get(test.Ctx(), curlPod.Name, metav1.GetOptions{})
+		g.Expect(err).NotTo(HaveOccurred())
+		return updatedCurlPod
+	}, TestTimeoutShort).Should(WithTransform(sampleyaml.IsPodRunningAndReady, BeTrue()))
+
+	// Check if the head pod is ready
+	g.Eventually(HeadPod(test, rayCluster), TestTimeoutShort).Should(WithTransform(sampleyaml.IsPodRunningAndReady, BeTrue()))
+
+	// Check if all worker pods are ready
+	g.Eventually(WorkerPods(test, rayCluster), TestTimeoutShort).Should(WithTransform(sampleyaml.AllPodsRunningAndReady, BeTrue()))
```

Comment:
Move these lines in between line 48 and line 50 (before creating the curl pod) to improve readability.